### ISSUE TITLE
Formula cquery depends on llvm@6 which is uninstallable

### ISSUE THREAD

**Issue by root42** [Linuxbrew/homebrew-core#10047]
- [x] have a problem with `brew install` (or `upgrade`, `reinstall`) a single formula? If it's a general `brew` problem please file this issue at Linuxbrew/brew: https://github.com/Linuxbrew/brew/issues/new/choose. If it's a tap (e.g. Brewsci/homebrew-bio) problem please file this issue at the tap.
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [ ] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [x] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

[brewconfig.txt](https://github.com/Linuxbrew/homebrew-core/files/2529946/brewconfig.txt)
[brewdoctor.txt](https://github.com/Linuxbrew/homebrew-core/files/2529947/brewdoctor.txt)

To help us debug your issue please explain:
- What you were trying to do (and why)

I installed cquery by running "brew install cquery". It installs latest llvm as a dependency (llvm@7).

- What happened (include command output)

cquery installs, but will not work:

```
$ cquery
cquery: error while loading shared libraries: libclang.so.6: cannot open shared object file: No such file or directory

```

- What you expected to happen

It seems the formula was linked against llvm 6.x, but it installs llvm 7.x as a dependency. Installing manually llvm@6 gives this:

```
$ brew install llvm@6
Error: undefined method `mac' for OS:Module
Did you mean?  mac?
Please report this bug:
  https://github.com/Linuxbrew/brew/wiki/troubleshooting
/home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/llvm@6.rb:20:in `block (2 levels) in <class:LlvmAT6>'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula_installer.rb:100:in `pour_bottle?'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula_installer.rb:496:in `expand_dependencies'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula_installer.rb:382:in `compute_dependencies'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula_installer.rb:143:in `verify_deps_exist'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula_installer.rb:136:in `prelude'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/cmd/install.rb:324:in `install_formula'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/cmd/install.rb:256:in `block in install'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/cmd/install.rb:254:in `each'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/cmd/install.rb:254:in `install'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/brew.rb:89:in `<main>'
```

- Step-by-step reproduction instructions (by running `brew install` commands)

```

brew install cquery
cquery
brew install llvm@6
```

**Comment by root42** [comment]
Reinstalling from source fixes this:

`brew reinstall --build-from-source cquery`

```
$ cquery -h
cquery is a low-latency C/C++/Objective-C language server.

Mode:
  --check <path>
...
```

**Comment by iMichka** [comment]
The `llvm@6` problem has been fixed with f3b523f269408f70dee5025d43523d36e4dd6e6b. Please run `brew update` and try again.

**Comment by iMichka** [comment]
I believe that this is now fixed. Closing here.
